### ISSUE TITLE
Interaction with identity server

### ISSUE THREAD

**Issue by TheKinrar** [blindnet-io/blindnet-backend-scala#3]
In our current production architecture, the backend has no Application model in its DB or anything like that, so it relies entirely on the identity server.
Therefore, on **each incoming request**, backend sends a request to identity server and the identity server performs the JWT verification. (Gateway, when it was in use, worked the same way)

This looks to me like an unnecessary overhead, and not a light one: each request to the backend will imply a request to the identity server that needs to be completed before the initial request is processed.

We could use some caching on the backend server, but right now identity is not providing any app info to backend, it's only able to check JWTs. An option would be to add an API to the identity server that lets the backend retrieve app info, cache it, and perform the JWT checks by itself. A probably better option would be to have the identity server push DB changes to the backend server, which would both remove the current overhead, and prevent any cache invalidation issues. Or a mix of these two options.

Any thoughts on this @m4rk055 @filipblnt?

**Comment by noelmace** [comment]
This should be discussed with @m4rk055 but with a low priority as DevKit is the top one.

**Comment by TheKinrar** [comment]
This was actually discussed on Slack. We decided on doing this for now:

> An option would be to add an API to the identity server that lets the backend retrieve app info, cache it, and perform the JWT checks by itself.

And then adding a Redis cache to improve performance.

Development was staled, waiting for https://github.com/blindnet-io/identity-api/issues/4.

Anyway this does not matter anymore as we're not deploying this backend and instead plan on replacing the existing one with the devkit components over time. We'll see if we can reuse some parts of this.
